gkctl/main.c

```diff
@@ -45,24 +45,38 @@ static char doc[] = "Gatekeeper Client -- configure Gatekeeper via "
 static struct argp_option options[] = {
 	{"server-path",		's',	"FILE",		0,
 		"Path to Gatekeeper/Grantor's UNIX socket",	1},
+	{"conn-timeout",	't',	"TIMEOUT",	0,
+		"UNIX socket connect timeout, in seconds",	1},
 	{ 0 }
 };
 
 struct args {
 	const char *server_path;
 	const char *lua_script_path;
+	unsigned int connect_timeout;
 };
 
 static error_t
 parse_opt(int key, char *arg, struct argp_state *state)
 {
 	struct args *args = state->input;
+	unsigned long timeout;
 
 	switch (key) {
 	case 's':
 		args->server_path = arg;
 		break;
 
+	case 't':
+		errno = 0;
+		timeout = strtoul(arg, NULL, 10);
```

Comment:
Style: `timeout` is only used in this `case`, open a brace and move the declaration of `timeout` to this new block to tighten scope.